subprojects/docs/src/docs/userguide/gradleWrapper.adoc

```diff
@@ -162,23 +256,34 @@ You can download the `.sha256` file by clicking on one of the `sha256` links on
 * https://gradle.org/release-candidate
 * https://gradle.org/nightly
 
++++++
+<figure xmlns:xi="http://www.w3.org/2001/XInclude">
+    <title>The SHA-256 link on the installation page</title>
+    <imageobject>
+        <imagedata fileref="img/sha256-link-installation-page.png" width="200mm" />
+    </imageobject>
+</figure>
++++++
+
 The format of the file is a single line of text that is the SHA-256 hash of the corresponding zip file.
 
+===== Configuring checksum verification
+
 Add the downloaded hash sum to `gradle-wrapper.properties` using the `distributionSha256Sum` property or use `--gradle-distribution-sha256-sum` on the command-line.
 
 .Configuring SHA-256 checksum verification
 ====
-
 [source,properties]
-.`gradle-wrapper.properties`
+.`gradle/wrapper/gradle-wrapper.properties`
 ----
-
 distributionSha256Sum=371cb9fbebbe9880d147f59bab36d61eee122854ef8c9ee1ecf12b82368bcf10
-
 ----
 ====
 
-[[sec:unix_file_permissions]]
-=== Unix file permissions
+Gradle will report a build failure in case the configured checksum does not match the checksum found on the server for hosting the distribution. Checksum Verification is only performed if the configured Wrapper distribution hasn't been downloaded yet.
 
-The Wrapper task adds appropriate file permissions to allow the execution of the `gradlew` *NIX command. Subversion preserves this file permission. We are not sure how other version control systems deal with this. What should always work is to execute “`sh gradlew`”.
+++++
+<sample xmlns:xi="http://www.w3.org/2001/XInclude" dir="userguide/wrapper" id="wrapperChecksumVerificationFailure" title="Checksum verification failure">
```

Comment:
I don't think this example is really necessary, but I whine a lot if we keep it.

Reply:
I wouldn't say it's necessary. It would be interesting though to know how this plays out in practice. I personally hadn't look at it before and wanted to know so I added it.